src/profiler.cpp

```
@@ -1125,6 +1128,8 @@ Error Profiler::start(Arguments& args, bool reset) {
         _add_event_frame = args._output != OUTPUT_JFR;
         _add_thread_frame = args._threads && args._output != OUTPUT_JFR;
         _add_sched_frame = args._sched;
+        _add_cpu_frame = args._record_cpu;
```

Since `record_cpu` is only supported with PerfEvents, do you want to return an error when `record_cpu` is requested for any other engine?

---

yeah that sounds like a good idea